File: app/src/main/java/de/sneakpeek/data/MovieInfoGenre.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

public class MovieInfoGenre implements Parcelable {

    public final static Parcelable.Creator<MovieInfoGenre> CREATOR = new Creator<MovieInfoGenre>() {

        public MovieInfoGenre createFromParcel(Parcel in) {
            MovieInfoGenre instance = new MovieInfoGenre();
            instance.id = ((int) in.readValue((int.class.getClassLoader())));
            instance.name = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public MovieInfoGenre[] newArray(int size) {
            return (new MovieInfoGenre[size]);
        }

    };

    @Json(name = "id") public int id;
    @Json(name = "name") public String name;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
        dest.writeValue(name);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieInfoGenre.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieInfoGenre(@Expose @SerializedName("id") var id: Int = 0,
                          @Expose @SerializedName("name") var name: String?) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieInfoGenre> = object : Parcelable.Creator<MovieInfoGenre> {
            override fun createFromParcel(source: Parcel): MovieInfoGenre = MovieInfoGenre(source)
            override fun newArray(size: Int): Array<MovieInfoGenre?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readInt(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(name)
    }
}

